packages/typespec-go/src/tcgcadapter/clients.ts

```diff
@@ -354,48 +366,78 @@ export class clientAdapter {
       }
     }
 
-    let sdkResponse: tcgc.SdkType | undefined;
+    let sdkResponseType = sdkMethod.response.type;
+
+    if (!sdkResponseType) {
+      // method doesn't return a type, so we're done
+      return respEnv;
+    }
+
+    // workaround until https://github.com/Azure/typespec-azure/issues/124 is fixed
+    if (sdkResponseType.kind === 'model' && (sdkResponseType.name === 'OperationStatus' || sdkResponseType.name === 'ResourceOperationStatus')) {
+      return respEnv;
+    }
+    // end workaround
 
-    if (go.isLROMethod(method) && sdkMethod.response.type) {
-      // use the method response for LROs as it contains the "final response" type
-      sdkResponse = sdkMethod.response.type;
-    } else if (bodyResponses.length === 1) {
-      sdkResponse = bodyResponses[0];
-    } else if (bodyResponses.length > 1) {
-      throw new Error('any response NYI');
+    // for paged methods, tcgc models the method response type as an Array<T>.
+    // however, we want the synthesized paged response envelope as that's what Go returns.
+    if (sdkMethod.kind === 'paging') {
+      // grab the paged response envelope type from the first response
+      sdkResponseType = Object.values(sdkMethod.operation.responses)[0].type!;
     }
 
-    if (sdkResponse?.kind === 'model') {
-      // workaround until https://github.com/Azure/typespec-azure/issues/124 is fixed
-      if (sdkResponse.name === 'OperationStatus' || sdkResponse.name === 'ResourceOperationStatus') {
-        return respEnv;
+    // we have a response type, determine the content type
+    let contentType: go.BodyFormat = 'binary';
+    if (sdkMethod.kind === 'lro') {
+      // we can't grovel through the operation responses for LROs as some of them
+      // return only headers, thus have no content type. while it's highly likely
+      // to only ever be JSON, this will be broken for LROs that return text/plain
+      // or a binary response. the former seems unlikely, the latter though...??
+      // TODO: follow up with tcgc team about this
```

Comment:
Can you make this an issue so it doesn't get forgotten. Add issue ID here so we can remove when closed on.

Reply:
Done.